sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -888,6 +888,21 @@ class DataFrame private[sql](
   }
 
   /**
+   * Returns a new [[DataFrame]] without duplicates.
+   * @group dfops
+   */
+  def dropDuplicates(): DataFrame = dropDuplicates(this.columns)
+
+  /**
+   * Returns a new [[DataFrame]] without duplicates under the given columns.
+   * @group dfops
+   */
+  def dropDuplicates(subset: Seq[String]): DataFrame = {
```

Comment:
Suppose that `distinct` is as same as `dropDuplicates` for removing duplicate rows? If they are the same, which implementation is better? `GroupedData` or `Distinct` node?